File: java/oop/CustomerApp/src/com/techlabs/customer/Customer.java
package com.techlabs.customer;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Customer {
	private String firstName;
	private String lastName;
	private Date dob;
	private static int id;
	private int customerId;

	static {
		id = 100;
	}

	public Customer(String firstName, String lastName, String dob)throws ParseException {
		id += 1;
		this.customerId = Customer.id;
		this.firstName = firstName;
		this.lastName = lastName;
		this.dob = converToDate(dob);
	}

	public int getCustomerId() {
		return this.customerId;
	}

	public String getFirstName() {
		return this.firstName;
	}

	public String getLastName() {
		return this.lastName;
	}

	private Date converToDate(String dob) throws ParseException {
		Date birthDate = null;
		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
		birthDate = sdf.parse(dob);
		return birthDate;
	}

	public String getBirthDate() {
		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
		return sdf.format(this.dob);
	}

	public int calculateAge() {
		int currentYear, birthYear, age;
		Date today = new Date();
		SimpleDateFormat sdf = new SimpleDateFormat("yyyy");
		String todaydate = sdf.format(today);
		currentYear = Integer.parseInt(todaydate);

		// System.out.println(todaydate);
		String birthdate = sdf.format(this.dob);
		birthYear = Integer.parseInt(birthdate);
		age = currentYear - birthYear;
		return age;
		// System.out.println(birthdate);
	}

}

File: dotNet/csharp/CustomerApp/CustomerApp/Customer.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace CustomerApp
{
    class Customer
    {
        private string _firstname;
        private string _lastname;
        private DateTime _dob;
        private static int _id;
        private int _customerId;

        static Customer()
        {
            _id = 100;
        }

        public Customer(string firstname,string lastname,string dob)
        {
            _id += 1;
            this._customerId = Customer._id;
            this._firstname = firstname;
            this._lastname = lastname;
            this._dob = ConvertToDate(dob);
        }

        public string Firstname
        {
            get
            {
                return _firstname;
            }
        }

        public string Lastname
        {
            get
            {
                return _lastname;
            }
        }

        public int CustomerId
        {
            get
            {
                return _customerId;
            }
        }
        private DateTime ConvertToDate(string dob)
        {

            DateTime birthdate = Convert.ToDateTime(dob);
            
         
            return birthdate;        
        }

        public string Birthdate
        {
            get
            {
                return _dob.ToLongDateString();
            }
        }

        public int CalculateAge()
        {
            int currentyear, birthyear, age;  
            currentyear= DateTime.Now.Year;
            birthyear = this._dob.Year;
            age = currentyear - birthyear;
            return age;
        }
    }
}
